contexts/RefetchContext.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { useRecoilValue } from 'recoil';
+import useSynthetixQueries from '@synthetixio/queries';
+
+import useGetFuturesOpenOrders from 'queries/futures/useGetFuturesOpenOrders';
+import useGetFuturesPositionForMarket from 'queries/futures/useGetFuturesPositionForMarket';
+import useGetFuturesMarketPositionHistory from 'queries/futures/useGetFuturesMarketPositionHistory';
+import { walletAddressState } from 'store/wallet';
+import useGetFuturesMarket from 'queries/futures/useGetFuturesMarket';
+import useGetFuturesPotentialTradeDetails from 'queries/futures/useGetFuturesPotentialTradeDetails';
+
+type RefetchType = 'modify-position' | 'new-order' | 'close-position' | 'margin-change';
+
+type RefetchContextType = {
+	handleRefetch: (refetchType: RefetchType, timeout?: number) => void;
+};
+
+const RefetchContext = React.createContext<RefetchContextType>({
+	handleRefetch: () => {},
+});
+
+export const RefetchProvider: React.FC = ({ children }) => {
+	const { useSynthsBalancesQuery } = useSynthetixQueries();
+	const walletAddress = useRecoilValue(walletAddressState);
+
+	const synthsBalancesQuery = useSynthsBalancesQuery(walletAddress);
+	const openOrdersQuery = useGetFuturesOpenOrders();
+	const positionQuery = useGetFuturesPositionForMarket();
+	const positionHistoryQuery = useGetFuturesMarketPositionHistory();
+	const marketQuery = useGetFuturesMarket();
+	useGetFuturesPotentialTradeDetails();
+
+	const handleRefetch = (refetchType: RefetchType, timeout?: number) => {
+		setTimeout(() => {
+			switch (refetchType) {
+				case 'modify-position':
+					marketQuery.refetch();
+					positionHistoryQuery.refetch();
```

Comment:
Yeah. We might be able to look into more fine-tuned polling. What do you think @Tburm. Part of the reason we moved to this new system for refetching is to avoid too much polling though.

However, I'm not sure this new state management/refetch pattern will be very useful in the unlikely event that subraph/contract data does not update before we call `handleRefetch`.

Reply:
Yeah. I think we have to look deeper into fixing the underlying issue here. Indefinite polling can't work as a long-term solution, so we probably need a more robust way of handling updates. It would be much easier if we could just have GraphQL subscriptions on the subgraph directly, but I don't think The Graph currently supports that officially (I think they had a couple of issues with the feature and pulled it). At the same time, I'm slightly worried that it might be a little out of scope here. What do you think about us creating a new issue that addresses this and plan a sync with @Tburm and other frontend devs?